### DISCUSSION TITLE
V8 - Pin row to the top of the table, regardless of sorting

### DISCUSSION THREAD

**Discussion by george-van-heerden** [TanStack/table#5039] [Q&A]
Firstly I'd like to say thanks for the wonderful migration guide, it helped me to migrate our tables from V7 with relative ease.

All our tables are relatively simple, but we have one table that requires the top record to stay at the top regardless of sorting.
Previously with version 7 we used custom sorting functions. When defining a column we used the sortType function to pass a custom sorting function:
```
// rest of column definition

sortType: (rowA, rowB, _, desc) => {
          //sort from a-z and always put creator at top of the list

          const a = rowA.original;
          const b = rowB.original;

          if (a.isCreator) {
            return desc ? 1 : -1;
          } else if (b.isCreator) {
            return desc ? -1 : 1;
          } else {
            return a.email > b.email ? 1 : -1;
          }
        },
```
It was pretty hacky, and I didn't like the solution, but it got the job done. Now with V8 we aren't able to use the same method since the sorting functions no longer have a prop to determine the sorting direction:
```
export type SortingFn<TData extends AnyData> = {
  (rowA: Row<TData>, rowB: Row<TData>, columnId: string): number
}
```
According to [this discussion](https://github.com/TanStack/table/discussions/4653), it is a feature that is going to be implemented eventually.

Having said all of this, I have three questions:
1. Has development for this feature started, and if so, when would it become available? (Happy to attempt to add or contribute to this feature)
2. Is there a different (better) approach that we could use to achieve the same result?
3. Is it possible to access the sorted direction from a custom sorting function?

**Comment by zahidiqbalnbs** [comment]
@george-van-heerden  any update on this please?
